plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -35,4 +45,162 @@ def ProxyShapePayloads(proxyShape, loadAll):
+    # in the layer stack is dirty.
+    if proxyStage:
+        # Is any layer in the layerstack dirty?
+        for layer in proxyStage.GetLayerStack(includeSessionLayers=False):
+            debugMessage('  Proessing layer: %s' % layer.GetDisplayName())
+            if layer.dirty:
+                debugMessage('    Found dirty layer')
+                return True
+    return False
+
+def GetStageFromProxyShapeAttr(attr):
+    # Helper method which returns the proxy shape name and Usd stage
+    # from the input attribute on the proxy shape.
+
+    # First get the stage name from the input attribute.
+    stageName = mel.eval('plugNode("%s")' % attr)
+
+    # Convert that into a long Maya path so we can get the USD stage.
+    res = cmds.ls(stageName, l=True)
+    fullStageName = res[0]
+    proxyStage = mayaUsd.ufe.getStage(fullStageName)
+
+    return(stageName, proxyStage)
+
+def ProxyShapeFilePathChanged(filePathAttr, newFilePath=None):
+    # Function called from the MayaUsd Proxy Shape template when the file path
+    # text field of the file path attibute custom control is modified or interacted with.
+    #
+    # Arguments:
+    # - filePathAttr: string representing the file path attribute.
+    # - newFilePath : If None, the user clicked the file browser button.
+    #                 - In this case a file dialog will be opened to allow user to pick USD file.
+    #                 Otherwise when a string, the user entered a filename directly in the field.
+    #                 - In this case simply try and load the file.
+    #
+    # Return Value:
+    # - True:  The input file path attribute was updated with a new value.
+    # - False: No update to file path attribute was done.
+    #
+    debugMessage('ProxyShapeFilePathChanged(%s, %s)' % (filePathAttr, newFilePath))
+    try:
+        stageName, proxyStage = GetStageFromProxyShapeAttr(filePathAttr)
+
+        # Does the filepath attribute contain a valid file?
+        currFilePath = cmds.getAttr(filePathAttr)
+        dirtyStack = False
+        openFileDialog = False
+        if currFilePath:
+            # We have a current file path, so check if any layer in the layerstack is dirty?
+            dirtyStack = IsProxyShapeLayerStackDirty(proxyStage)
+            if not dirtyStack:
+                openFileDialog = True
+        else:
+            # No current file path, so do we just have an empty stage or was something added to
+            # this anon layer?
+            rootLayer = proxyStage.GetRootLayer()
+            if rootLayer and not rootLayer.empty:
+                dirtyStack = True
+            elif newFilePath is None:
+                debugMessage('  Empty stage, opening file dialog...')
+                openFileDialog = True
+
+        # If we have a dirty layer stack, display dialog to confirm new loading.
+        if dirtyStack:
+            kTitleFormat = mel.eval('getMayaUsdString("kDiscardStageEditsTitle")')
```

Comment:
Pretty awkward that there is no way to do this from Python.  Certainly getMayaUsdString() isn't doing much, and 
https://git.autodesk.com/maya3d/maya/blob/master/Maya/src/OpenMaya/Commands/getPluginResource.mel
isn't doing much either...  I guess we can just leave it as you have it, not a big deal.

Reply:
I looked at the MEL procs for register/get string resources and unfortunately they don't have python equivalents. But they are trivial, so I've implemented a python version of them in our plugin and now no longer need the mel.eval.